src/validate.js

```diff
@@ -102,23 +107,214 @@ class ApplyRules {
     return errors;
   }
 
-  static applyModelRules(rules, nodeToTest) {
+  /**
+   * @deprecated since version 1.2.0, since it uses synchronous IO
+   *   (https://nodejs.org/en/docs/guides/blocking-vs-non-blocking/#comparing-code). Use
+   *   applyModelRulesAsync() instead
+   */
+  static applyModelRulesSync(rules, nodeToTest) {
     let errors = [];
     for (const rule of rules) {
       // Apply whole-model rule, and field-specific rules
-      errors = errors.concat(rule.validate(nodeToTest));
+      errors = errors.concat(rule.validateSync(nodeToTest));
     }
     for (const field in nodeToTest.value) {
       if (Object.prototype.hasOwnProperty.call(nodeToTest.value, field)) {
         // If this field is itself a model, apply that model's rules to it
-        errors = errors.concat(this.applySubModelRules(rules, nodeToTest, field));
+        errors = errors.concat(this.applySubModelRulesSync(rules, nodeToTest, field));
       }
     }
     return errors;
   }
+
+  static async applySubModelRulesAsync(rules, nodeToTest, field) {
```

Comment:
This is almost identical to the previous `applySubModelRules` method except uses `await` where appropriate